labman/gui/static/js/plate.js

```diff
@@ -116,5 +116,52 @@ function change_plate_configuration() {
     // reset the container before updating the grid configuration
     $('#plate-map-div').empty().height(0);
     pv = new PlateViewer('plate-map-div', undefined, undefined, $opt.attr('pm-data-rows'), $opt.attr('pm-data-cols'));
+    // Enable the plate create button now that a plate config is selected
+    $('#createPlateBtn').prop('disabled', false);
   }
 }
+
+/**
+ * Callback from the "Create" button in the sample plating page.
+ */
+function createPlate(){
+  $('#createPlateBtn').prop('disabled', true);
+
+  var plateName = $('#newNameInput').val().trim();
+  var plateConf = $('#plate-conf-select option:selected').val();
+
+  $.post(
+    '/process/sample_plating',
+    {'plate_name': plateName, 'plate_configuration': plateConf}
+  )
+  .done(function (data) {
+    var plateId = data['plate_id'];
+    var processId = data['process_id'];
+
+    $('#plateName').prop('pm-data-plate-id', plateId);
+    $('#plateName').prop('pm-data-process-id', processId);
+    // Once the plate has been created, we can disable the plate config select
+    $('#plate-conf-select').prop('disabled', true);
+
+    // reset the container before updating the grid configuration
+    $('#plate-map-div').empty().height(0);
+    var $opt = $('#plate-conf-select option:selected');
+    var pv = new PlateViewer('plate-map-div', undefined, undefined, $opt.attr('pm-data-rows'), $opt.attr('pm-data-cols'));
+    pv.plateId = plateId;
```

Comment:
I realized that there had better be a comment on these lines, or the next time I see them, I will try to "fix" them the way I did yesterday (put the plateId and the processId into the two undefined values in the "new PlateViewer" call), which I currently remember doesn't work.  Could you add this comment above these lines?

// Create a new PlateViewer object *without* passing in plateId and processId,
// because if those are passed in to the object initializer, code assumes the  
// plate has already been filled--and when it can't find contents, it populates
// all the wells with blanks :(.  Setting the plateId and processId after object
// initialization sets the necessary state but avoids this behavior.

Reply:
👍 should be added now.